internal/database/metadata/postgres/feature.go

```diff
@@ -102,3 +103,68 @@ func getFeatureByName(ctx context.Context, sqlxCtx metadata.SqlxContext, name st
 
 	return &feature, nil
 }
+
+func listFeature(ctx context.Context, sqlxCtx metadata.SqlxContext, opt metadata.ListFeatureOpt) (types.FeatureList, error) {
+	var (
+		features types.FeatureList
+		err      error
+	)
+
+	query := `SELECT * FROM "feature"`
+	cond, args, err := buildListFeatureCond(opt)
+	if err != nil {
+		return nil, err
+	}
+	if len(cond) > 0 {
+		query = fmt.Sprintf("%s WHERE %s", query, strings.Join(cond, " AND "))
+	}
+	if err := sqlxCtx.SelectContext(ctx, &features, sqlxCtx.Rebind(query), args...); err != nil {
+		return nil, err
+	}
+
+	// enrich group
+	groupIDs := features.GroupIDs()
+	groups, err := listGroup(ctx, sqlxCtx, nil, &groupIDs)
+	if err != nil {
+		return nil, err
+	}
+	for _, f := range features {
+		group := groups.Find(func(g *types.Group) bool { return g.ID == f.GroupID })
+		if group == nil {
+			return nil, errdefs.InvalidAttribute(fmt.Errorf("no group found for feature %s", f.Name))
+		}
+		f.Group = group
+	}
+
+	// filter by entity
+	if opt.EntityID != nil && !features.All(func(f *types.Feature) bool {
+		return f.Group.EntityID == *opt.EntityID
+	}) {
+		return nil, nil
+	}
+	return features, nil
```

Comment:
does it filter out any features?

Reply:
Yes, if `opt.EntityID ! = nil` and has a feature exists `feature.Group.EntityID != opt.EntityID` returns nil

Reply:
hmm shouldn't we return other features that `feature.Group.EntityID == opt.EntityID`?

Reply:
my wrong 🥶  , done.